pkg/cloud/azure/storageconnection.go

```diff
@@ -80,11 +81,18 @@ func (sc *StorageConnection) StreamBlob(blobName string, client *azblob.Client)
 }
 
 // DownloadBlobToFile downloads the Azure Billing CSV to a local file
-func (sc *StorageConnection) DownloadBlobToFile(localFilePath string, blobName string, client *azblob.Client, ctx context.Context) error {
-	// If file exists, don't download it again
-	if _, err := os.Stat(localFilePath); err == nil {
-		log.DedupedInfof(3, "CloudCost: Azure: DownloadBlobToFile: file %v already exists, not downloading %v", localFilePath, blobName)
-		return nil
+func (sc *StorageConnection) DownloadBlobToFile(localFilePath string, blob container.BlobItem, client *azblob.Client, ctx context.Context) error {
+	blobName := *blob.Name
+	// Check if file already exists
+	if fileInfo, err := os.Stat(localFilePath); err == nil {
+		blobModTime := *blob.Properties.LastModified
+		// Check if the blob was last modified before the file was modified, indicating that the
+		// file is the most recent version of the blob
+		if blobModTime.Before(fileInfo.ModTime()) {
```

Comment:
What happens if this is not true? In other words, if the Blob file has a more recent timestamp than our local file's timestamp? Does this mean we need to go fetch the updated CSV file from storage?

^^ I don't think that scenario would ever happen though.

Reply:
Oops nevermind. After re-reading this, I see that we would proceed to download the updated Blob file! This comment is good to close.